internal/discord/commands.go

```diff
@@ -121,7 +267,87 @@ func (d *Discord) Exile(s *discordgo.Session, i *discordgo.InteractionCreate) {
 //	user:		User
 //	reason:		string
 func (d *Discord) Unexile(s *discordgo.Session, i *discordgo.InteractionCreate) {
-	return
+	isFirst := true
+	optionMap := mapOptions(i)
+	exileRole := d.Roles[i.GuildID]["Exiled"]
+	verifiedRole := d.Roles[i.GuildID]["Verified"]
+	logMsg, err := d.LogCommand(i.Interaction)
+	if err != nil {
+		fmt.Printf("Failed to log: %v\n", err)
+	}
+
+	exiledUser := optionMap["user"].UserValue(nil)
+
+	// Check if user exists in guild
+	exiledMember, err := d.CheckUserInGuild(i.GuildID, exiledUser.ID)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not find user <@%v> in guild", exiledUser.ID)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		RespondToInteraction(s, i.Interaction, tempstr, &isFirst)
+		return
+	}
+
+	// Check if user has specified roles
+	isExiled := false
+	isVerified := false
+	for _, role := range exiledMember.Roles {
+		if role == exileRole.ID {
+			isExiled = true
+		} else if role == verifiedRole.ID {
+			isVerified = true
+		}
+	}
+
+	if !isExiled {
+		tempstr := fmt.Sprintf("User <@%v> is not currently exiled, nothing has been done", exiledUser.ID)
+		RespondToInteraction(s, i.Interaction, tempstr, &isFirst)
+		logMsg.Embeds[0].Description += fmt.Sprintf("\n%v", tempstr)
+		_, err = d.Session.ChannelMessageEditEmbed(d.ModLoggingChannelID, logMsg.ID, logMsg.Embeds[0])
+		if err != nil {
+			fmt.Printf("Unable to edit log message: %v\n", err)
+		}
+		return
+	}
+
+	if isVerified {
+		tempstr := fmt.Sprintf("User <@%v> is both exiled and verified, nothing has been done", exiledUser.ID)
+		RespondToInteraction(s, i.Interaction, tempstr, &isFirst)
+		logMsg.Embeds[0].Description += fmt.Sprintf("\n%v", tempstr)
+		_, err = d.Session.ChannelMessageEditEmbed(d.ModLoggingChannelID, logMsg.ID, logMsg.Embeds[0])
+		if err != nil {
+			fmt.Printf("Unable to edit log message: %v\n", err)
+		}
+		return
+	}
```

Comment:
how you have it right now is hard to combine with the helper func i suggested in `Exile`. i think we should move this to its own error checking func but it would be even better if we could merge the two into one general error checking func

Reply:
I've been thinking about this for a bit and I decided the way to go about this is to just move this logic (and the similar-ish logic for `/exile`)  into the `ExileUser` and `UnexileUser` helper functions. My reasoning behind these is that the two actions should be paired as an atomic action since I'm always going to check for existing roles before updating the user's roles.